bin/node/runtime/src/impls.rs

```diff
@@ -89,24 +89,33 @@ impl ProposalProvider<AccountId, Hash, RuntimeCall> for AllianceProposalProvider
 
 	fn vote_proposal(
 		who: AccountId,
-		proposal: Hash,
+		proposal_bounded: Bounded<RuntimeCall>,
```

Comment:
I actually asked Gav whether this is the intended usage of `Bounded`; turns out it is not. He said it does not make sense to pass the proposal itself, but rather the hash. So we need a map again for `Hash -> Bounded<Proposal>`.

Reply:
A `Hash -> Bounded<Proposal>` map with a `Bounded<Proposal> -> Proposal` given by pallet preimages doesn't make any sense, I'll revert to a `Hash -> Proposal` inside pallet collective